roll.rb

```diff
@@ -0,0 +1,75 @@
+# Three 1's => 1000 points
+# Three 6's =>  600 points
+# Three 5's =>  500 points
+# Three 4's =>  400 points
+# Three 3's =>  300 points
+# Three 2's =>  200 points
+# One   1   =>  100 points
+# One   5   =>   50 points
+
+class Roll
+	attr_reader :no_of_dices, :outcome, :non_scoring_dices, :score
+
+	def initialize(no_of_dices, options = {})
+		@no_of_dices = no_of_dices
+		@outcome = Array.new
+		@score = 0
+		simulate_roll(options)
+		calc_score()
+	end
+
+	def all_scoring_dices?()
+		non_scoring_dices.empty?
+	end
+
+	def to_s()
+		"Rolled #{no_of_dices} dices. \n" +
+		"Outcome is #{outcome.join(',')} \n" +
+		"Roll score is #{score} \n" +
+		"You have #{non_scoring_dices.count} non-scoring dices. \n"
+	end
+
+	private
+
+	def simulate_roll(options)
+		unless options[:override_outcome]
+			no_of_dices.times do
+				@outcome << (1 + rand(6))
+			end
+		else
+			@outcome = options[:override_outcome].clone
+		end
+
+		@non_scoring_dices = @outcome.clone
+	end
+
+	def calc_score()
+		# check for triplets of 1,2,3,4,5,6
+		[1,6,5,4,3,2].each do |number|
+			if arr_has_three_of_a_kind?(@outcome, number)
+				temp_score = (number == 1) ? 1000 : (number*100)
```

Comment:
Leave spaces around operators
```suggestion
				temp_score = (number == 1) ? 1000 : (number * 100)
```

Reply:
done, committed